api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+                                                "schema": {
+                                                        "$ref": "#/definitions/NodeConfigurationResponse"
+                                                }
+					}
+				}
+			},
+                        "put": {
+                                "summary": "Update daemon configuration",
+				"tags": [ "daemon" ],
+				"parameters": [
+					{
+						"name": "configuration",
+						"in": "body",
+						"required": true,
+						"schema": {
+							"$ref": "#/definitions/ConfigurationMap"
+						}
+					}
+				],
+                                "responses": {
+					"200": {
+						"description": "Successful update"
+					},
+                                        "default": {
+                                                "description": "Unexpected error",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+                                }
+                        }
+                },
+                "/endpoint": {
+			"get": {
+				"summary": "Retrieve an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"put": {
+				"summary": "Create or update an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "endpoint",
+						"in": "body",
+						"required": true,
+						"schema": {
+							"$ref": "#/definitions/Endpoint"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "Success",
+						"schema": {
+							"$ref": "#/definitions/EndpointResponse"
+						}
+					},
+                                        "405": {
+                                                "description": "Endpoint already exists",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        },
+                                        "default": {
```

Comment:
It currently falls under default but this is a good case for an error that must be distinguishable